### DISCUSSION TITLE
minimal mode valid attributes

### DISCUSSION THREAD

**Discussion by 440bx** [zyantific/zydis#527] [Q&A] [answered]
Hello, 

the description of minimal mode  (the one I found) states that some instruction attributes are set, e.g, ZYDIS_ATTRIBUTE_IS_RELATIVE while others are not (the description states "most" are not.)

Is there a list somewhere (other than reading the source code) of the attributes that are set in minimal mode ? (iow, can be relied upon.)  

Thank you for your help.

**Comment by flobernd** [comment, marked as answer]
Hi @440bx ,

it's not documented anywhere currently. From memory, all the encoding related ones should be available:

- `HAS_MODRM`
- `HAS_SIB`
- ...

besides that, all the segment ones:

- `HAS_SEGMENT_CS`
- HAS_SEGMENT_DS`
- ...

and the mentioned `IS_RELATIVE` plus `ACCEPTS_LOCK`/`HAS_LOCK`.

This should be everything.

**Reply by 440bx** [reply]
Thank you.

If I may, I'd like to suggest that it would be nice to have the attributes that can be relied upon in minimal mode, documented.
